### PR TITLE
docs(framework:skip) Update Docker Compose docs for 1.13.0

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

**Review by chongshenng — DISMISSED** [review]
lgtm!

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 1]
```diff
@@ -6,8 +6,8 @@ Compose.
 
 You will learn how to run the Flower client and server components on two separate
 machines, with Flower configured to use TLS encryption and persist SuperLink state
-across restarts. A server consists of a SuperLink and ``SuperExec``. For more details
-about the Flower architecture, refer to the :doc:`../explanation-flower-architecture`
+across restarts. A server consists of a SuperLink and ServerApp. For more details about
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:9]:
```suggestion
across restarts. A server consists of a SuperLink and a ``ServerApp``. For more details about
```

File: doc/source/docker/tutorial-quickstart-docker-compose.rst [hunk 2]
```diff
@@ -43,7 +43,7 @@ Step 1: Set Up
        $ export PROJECT_DIR=quickstart-compose
 
    Setting the ``PROJECT_DIR`` helps Docker Compose locate the ``pyproject.toml`` file,
-   allowing it to install dependencies in the SuperExec and SuperNode images correctly.
+   allowing it to install dependencies in the ClientApp and ServerApp images correctly.
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker-compose.rst:46]:
```suggestion
   allowing it to install dependencies in the ``flwr/serverapp` and ``flwr/clientapp`` images correctly.
```

Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker-compose.rst:46]:
```suggestion
   allowing it to install dependencies in the ServerApp and ClientApp images correctly.
```

File: doc/source/docker/tutorial-quickstart-docker-compose.rst [hunk 3]
```diff
@@ -321,36 +309,71 @@ In ``with-tls.yml``, add the following:
...
-           #   - --database=state/state.db
+           # - --insecure
+           # - --executor
+           # - flwr.superexec.deployment:executor
+           # - --isolation
+           # - process
+           # - --database=state/state.db
            command:
+             - --executor
+             - flwr.superexec.deployment:executor
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker-compose.rst:374]: Let's remove these lines since it's `flwr.superexec.deployment:executor` by default.

**Review by chongshenng — CHANGES_REQUESTED** [review]
@Robert-Steiner thanks! Overall looks good, only minor comments.

**Review by chongshenng — APPROVED** [review]
LGTM!

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 4]
```diff
@@ -6,7 +6,7 @@ Compose.
 
 You will learn how to run the Flower client and server components on two separate
 machines, with Flower configured to use TLS encryption and persist SuperLink state
-across restarts. A server consists of a SuperLink and ``SuperExec``. For more details
+across restarts. A server consists of a SuperLink and a ServerApp. For more details
```
Comment (danieljanes) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:9]:
```suggestion
across restarts. A server consists of a SuperLink and a ``ServerApp``. For more details
```

**Review by danieljanes — APPROVED** [review]

**Pull request merged**